web-service/shortener.py

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python
+#
+# Copyright 2015 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import webapp2
+from google.appengine.api import urlfetch
+
+################################################################################
+
+class ShortURL(webapp2.RequestHandler):
+    def post(self):
+        input_data = self.request.body
+        with open('API_KEYS.json') as keysfile:
+            apikey = json.load(keysfile)["goo.gl"]
+        url = 'https://www.googleapis.com/urlshortener/v1/url?key=' + apikey
+        referer = 'url-cast.physical-web.org'
+        result = urlfetch.fetch(url,
+                validate_certificate=True,
+                method=urlfetch.POST,
+                payload=input_data,
+                headers = {
+                    'Content-Type': 'application/json',
+                    'Referer': referer
+                })
+        self.response.headers['Content-Type'] = 'application/json'
+        self.response.write(result.content)
+
+################################################################################
+
+app = webapp2.WSGIApplication([
+    ('/shorten-url', ShortURL),
+], debug=True)
```

Comment:
This I took from the existing handlers.py, but looking at the docs, its:

> The parameter debug=true passed to its constructor tells webapp2 to print stack traces to the browser output if a handler encounters an error or raises an uncaught exception. You may wish to remove this option from the final version of your application.

We should probably turn this off before shipping to prod, but given that its open source I don't think its actually harmful either way.  I suspect we've left this True since the start!